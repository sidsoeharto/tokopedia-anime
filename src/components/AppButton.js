/** @jsxImportSource @emotion/react */

import React from "react";
import AppColors from "../styles/AppColors";

const AppButton = ({ title, onClick, style, icon }) => {
  return (
      <button
        onClick={onClick}
        css={{
          padding: 8,
          borderRadius: 8,
          backgroundColor: AppColors.pink400,
          borderWidth: 0,
          alignItems: 'center',
          margin: '8px 0px',
          cursor: 'pointer',
          ...style
        }}
      >
        {icon}
        <span css={{color: AppColors.gray100, fontWeight: 'bold'}}>{title}</span>
      </button>
  );
};

export default AppButton;