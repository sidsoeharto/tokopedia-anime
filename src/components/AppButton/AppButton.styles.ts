import { css } from '@emotion/react';
import AppColors from '../../styles/AppColors';

export const appButton = css`
  padding: 8px;
  border-radius: 8px;
  background-color: ${AppColors.pink400};
  border-width: 0;
  align-items: center;
  margin: 8px 0px;
  cursor: pointer;
  font-family: Overpass;
`;

export const buttonText = css`
  color: ${AppColors.gray100}; 
  font-weight: bold;
`