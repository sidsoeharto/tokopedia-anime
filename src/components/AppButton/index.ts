export { default } from "./AppButton.component";