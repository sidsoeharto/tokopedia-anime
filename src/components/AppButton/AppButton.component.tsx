/** @jsxImportSource @emotion/react */

import React, { ReactNode } from "react";
import { appButton, buttonText } from "./AppButton.styles";

interface AppButtonProps {
  title?: string;
  onClick: () => void;
  className?: string;
  icon?: ReactNode;
}

const AppButton: React.FC<AppButtonProps> = ({
  title,
  onClick,
  className,
  icon,
}) => {
  return (
    <button
      data-testid="app-button"
      onClick={onClick}
      css={appButton}
      className={className}
    >
      {icon}
      <span css={buttonText}>{title}</span>
    </button>
  );
};

export default AppButton;