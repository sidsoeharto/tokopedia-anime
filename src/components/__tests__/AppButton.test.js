import { createRoot } from 'react-dom/client';
import renderer from "react-test-renderer";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { createSerializer } from '@emotion/jest'

import AppButton from '../AppButton';

expect.addSnapshotSerializer(createSerializer())

afterEach(cleanup)

describe('AppButton Test', () => {
  it('renders button correctly', () => {
    const {getByTestId} = render(<AppButton title={'Test Button'}/>)
    expect(screen.getByTestId('app-button')).toHaveTextContent('Test Button');
  })
  it('should matches snapshot', () => {
    const tree = renderer.create(
      <AppButton 
        title={'test'}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})