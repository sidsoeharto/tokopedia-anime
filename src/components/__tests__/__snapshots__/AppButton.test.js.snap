// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton Test should matches snapshot 1`] = `
.emotion-0 {
  padding: 8px;
  border-radius: 8px;
  background-color: #f472b6;
  border-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px 0px;
  cursor: pointer;
  font-family: Overpass;
}

.emotion-1 {
  color: #fafafa;
  font-weight: bold;
}

<button
  className="emotion-0"
  data-testid="app-button"
>
  <span
    className="emotion-1"
  >
    test
  </span>
</button>
`;
